### PR TITLE
Code early stages

### PR DESCRIPTION
**Pull request by Trey50Daniel**
Added beginning movement + camera script to follow the player along the level. Player can move left or right on their journey to the end goal.

### CODE REVIEW

File: Assets/Scripts/CameraScript.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class CameraScript : MonoBehaviour
+{
+
+    public GameObject player;       //Public variable to store a reference to the player game object
```
Comment (kinbiko) [inline Assets/Scripts/CameraScript.cs]: Redundant comment, remove.

Comment (kinbiko) [inline Assets/Scripts/CameraScript.cs]: Does the player really need to be public?

File: Assets/Scripts/CameraScript.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
```
Comment (kinbiko) [inline Assets/Scripts/CameraScript.cs]: At most 1 line between fields please.

File: Assets/Scripts/CameraScript.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    private Vector3 offset;         //Private variable to store the offset distance between the player and camera
```
Comment (kinbiko) [inline Assets/Scripts/CameraScript.cs]: Redundant comment, instead rename variable to `playerCameraOffset`.

File: Assets/Scripts/CameraScript.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    // Use this for initialization
```
Comment (kinbiko) [inline Assets/Scripts/CameraScript.cs]: Redundant comment

File: Assets/Scripts/CameraScript.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class CameraScript : MonoBehaviour
+{
+
```
Comment (kinbiko) [inline Assets/Scripts/CameraScript.cs:8]: All comments in this file are redundant.

File: Assets/Scripts/CameraScript.cs [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    void Start()
+    {
+        //Calculate and store the offset value by getting the distance between the player's position and camera's position.
+        offset = transform.position - player.transform.position;
+    }
+
+    // LateUpdate is called after Update each frame
+    void LateUpdate()
+    {
+        // Set the position of the camera's transform to be the same as the player's, but offset by the calculated offset distance.
```
Comment (kinbiko) [inline Assets/Scripts/CameraScript.cs]: So this is the magic that stops the spinning camera? If so, comment just that.

File: Assets/Scripts/PlayerMovement.cs [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+public class PlayerMovement : MonoBehaviour {
+    private Rigidbody2D rgBody;
+    [SerializeField]
+    private float movementSpeed;
+
+    [SerializeField]
+    private Transform[] groundedPoints;
+
+	void Start () {
```
Comment (kinbiko) [inline Assets/Scripts/PlayerMovement.cs]: Use consistent indentation

Comment (kinbiko) [inline Assets/Scripts/PlayerMovement.cs]: and use

```
Start
{
//method body
}
```

style method signatures

File: Assets/Scripts/PlayerMovement.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        rgBody = GetComponent<Rigidbody2D>();
+	}
+	
+	void FixedUpdate () {
+        float horizontal = Input.GetAxis("Horizontal");
+        Debug.Log(horizontal);
```
Comment (kinbiko) [inline Assets/Scripts/PlayerMovement.cs]: Remove

File: Assets/Scripts/PlayerMovement.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+        movementHandler(horizontal);
+	}
+
+    private void movementHandler(float horizontal)
+    {
+        rgBody.velocity = new Vector2(horizontal * movementSpeed, rgBody.velocity.y);
+
+        if(Input.GetButtonDown("Jump") && rgBody.velocity.y == 0)
```
Comment (kinbiko) [inline Assets/Scripts/PlayerMovement.cs]: extract to private method `shouldJump()`

File: Assets/Scripts/PlayerMovement.cs [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            rgBody.velocity = new Vector2(horizontal * movementSpeed, 3);
+        }
+    }
+/*
+    private bool isGrounded()
```
Comment (kinbiko) [inline Assets/Scripts/PlayerMovement.cs]: No commented code please

File: Assets/Scripts/PlayerMovement.cs [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    [SerializeField]
+    private Transform[] groundedPoints;
+
+	void Start ()
+    {
+        rgBody = GetComponent<Rigidbody2D>();
+	}
+	
+	void FixedUpdate ()
+    {
```
Comment (kinbiko) [inline Assets/Scripts/PlayerMovement.cs:18]: Indentation still not OK.

File: Assets/Scripts/PlayerMovement.cs [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+        Debug.Log(horizontal);
+
+        movementHandler(horizontal);
+	}
+
+    private void movementHandler(float horizontal)
+    {
+        rgBody.velocity = new Vector2(horizontal * movementSpeed, rgBody.velocity.y);
+
+        shouldJump(horizontal);
```
Comment (kinbiko) [inline Assets/Scripts/PlayerMovement.cs]: in general, methods starting with `should`/`is`/`can` etc. return booleans. So what I meant was to put the stuff inside the `if()` in a method, and leave the rest in the movementHandler.

File: Assets/Scripts/PlayerMovement.cs [hunk 13]
```diff
@@ -24,18 +26,14 @@ private void movementHandler(float horizontal)
...
-        if(Input.GetButtonDown("Jump") && rgBody.velocity.y == 0)
-        {
-            rgBody.velocity = new Vector2(horizontal * movementSpeed, 3);
-        }
+        shouldJump(horizontal);
     }
-/*
-    private bool isGrounded()
+
+    private void shouldJump(float horizontal)
```
Comment (burtybob) [inline Assets/Scripts/PlayerMovement.cs]: To me "ShouldJump" sounds like it's a boolean function?

File: Assets/Scripts/CameraScript.cs [hunk 14]
```diff
@@ -4,22 +4,21 @@
...
 {
 
-    public GameObject player;       //Public variable to store a reference to the player game object
+    public GameObject player;
 
 
-    private Vector3 offset;         //Private variable to store the offset distance between the player and camera
-
-    // Use this for initialization
+    private Vector3 offset;
```
Comment (burtybob) [inline Assets/Scripts/CameraScript.cs]: This would be better renamed as "playerCameraOffset"?

File: Assets/Scripts/CameraScript.cs [hunk 15]
```diff
@@ -0,0 +1,24 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class CameraScript : MonoBehaviour
+{
+
+    public GameObject player;
+
```
Comment (kinbiko) [inline Assets/Scripts/CameraScript.cs:8]: remove excessive empty lines

File: Assets/Scripts/CameraScript.cs [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+
+    private Vector3 offset;         
+   
+    void Start()
+    {
+
```
Comment (kinbiko) [inline Assets/Scripts/CameraScript.cs]: remove excessive empty line

**Pull request merged**